modules/openapi-generator/src/main/java/org/openapitools/codegen/SortByInheritanceFirstBuilder.java

```diff
@@ -0,0 +1,112 @@
+package org.openapitools.codegen;
+
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.media.Schema;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.openapitools.codegen.utils.ModelUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class SortByInheritanceFirstBuilder {
```

Comment:
would it be possible to add some unit tests for this new class, especially the reorder method?

Reply:
There is SpringCodegenTest.testAllArgsConstructor_inheritanceFirst

But you're right.  As this class is (nearly) a standalone class, we can unit test it.
I will do it.

Reply:
> There is SpringCodegenTest.testAllArgsConstructor_inheritanceFirst

I saw that too 👍 

for every new class, we want some unit tests to make sure things are good as a starting point before further modifying it down the road.